internal/plugin/server.go

```diff
@@ -198,17 +200,29 @@ func (plugin *nvidiaDevicePlugin) Serve() error {
 				break
 			}
 
-			klog.Infof("GRPC server for '%s' crashed with error: %v", plugin.rm.Resource(), err)
-
 			timeSinceLastCrash := time.Since(lastCrashTime).Seconds()
 			lastCrashTime = time.Now()
-			if timeSinceLastCrash > 3600 {
+
+			klog.Errorf(
+				"GRPC server for '%s' crashed with error: %v. Retry attempt: %d, Time since last crash: %.2f seconds",
+				plugin.rm.Resource(),
+				err,
+				restartCount+1,
+				timeSinceLastCrash,
+			)
+
+			if timeSinceLastCrash > crashTimeoutSeconds {
 				// it has been one hour since the last crash.. reset the count
 				// to reflect on the frequency
```

Comment:
Let's rather just log the `crashTimeout` instead of a comment that is now further from the place where the one hour is specified.